File: apps/rails/spec/factories/dividend_rounds.rb
# frozen_string_literal: true

FactoryBot.define do
  factory :dividend_round do
    company
    issued_at { 1.day.ago }
    number_of_shares { 100 }
    number_of_shareholders { 10 }
    total_amount_in_cents { 230_01 }
    status { "Issued" }
    return_of_capital { false }
  end
end

File: e2e/factories/dividendRounds.ts
import { faker } from "@faker-js/faker";
import { db } from "@test/db";
import { companiesFactory } from "@test/factories/companies";
import { dividendRounds } from "@/db/schema";
import { assert } from "@/utils/assert";

export const dividendRoundsFactory = {
  create: async (overrides: Partial<typeof dividendRounds.$inferInsert> = {}) => {
    const [insertedDividendRound] = await db
      .insert(dividendRounds)
      .values({
        companyId: overrides.companyId ?? (await companiesFactory.create()).company.id,
        issuedAt: faker.date.recent(),
        numberOfShares: 1000n,
        numberOfShareholders: 10n,
        totalAmountInCents: 100000n,
        status: "Issued",
        returnOfCapital: false,
        readyForPayment: true,
        releaseDocument: "This is a release agreement for {{investor}} for the amount of {{amount}}.",
        ...overrides,
      })
      .returning();
    assert(insertedDividendRound != null);

    return insertedDividendRound;
  },
};
